### PR TITLE
docs: create NavLink cookbook example

### PR DESCRIPTION
**Pull request by Adbib**
The NavLink component enhances navigation by getting isPending and isActive status to styling your links

# Overview

# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [x] Docs / tests / types / typos

# Description
When building navigation in Qwik, you may want to know when a link is active or pending navigation. The standard <a> and <Link> components don't provide this.

The NavLink component enhances navigation links by adding:
- **Active Status** - Apply a class when the link matches the current location. 
- **Pending Status** - Apply a class during navigation when linking to the target location.

# Use cases and why

- 1. navbar
- 2. sidebar

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Review by gioboa — CHANGES_REQUESTED** [review]
You need to add the new section here.
`packages/docs/src/routes/docs/menu.md`
`packages/docs/src/routes/docs/cookbook/index.mdx`
Can you update this files too pls?

File: packages/docs/src/routes/docs/cookbook/nav-link/index.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+</NavLink>
+```
+
+## How NavLink Works 
+
+Under the hood, NavLink uses the useLocation hook to get navigation status. It checks if the link href matches the current URL pathname to set isActive. It uses isNavigating to determine pending status when navigating to the link.
+
+This allows NavLink to know the active and pending state automatically based on navigation.
+
+```tsx
```
Comment (gioboa) [inline packages/docs/src/routes/docs/cookbook/nav-link/index.mdx:27]: Can you change this snippet with a component you can check the other examples with this imports
`import CodeSandbox, {CodeFile} from '../../../../components/code-sandbox/index.tsx';`
This helps us ensure alignment over time.

Comment (Adbib) [reply packages/docs/src/routes/docs/cookbook/nav-link/index.mdx:27]: sorry i don't understand

Comment (gioboa) [reply packages/docs/src/routes/docs/cookbook/nav-link/index.mdx:27]: The other cookbook examples are using `CodeSandbox` and `CodeFile` to ensure alignment over time.
Can you do the same pls?

Comment (Adbib) [reply packages/docs/src/routes/docs/cookbook/nav-link/index.mdx:37]: done

**Comment by gioboa** [comment]
> You need to add the new section here. `packages/docs/src/routes/docs/menu.md` `packages/docs/src/routes/docs/cookbook/index.mdx` Can you update this files too pls?

Can you do the change in this PR and not in a different one pls?

**Comment by gioboa** [comment]
@Adbib I fixed up the docs. I added my suggestions too.
Are we considering all the cases for isActive?
what about href="/about/" 
is it considered?

**Comment by Adbib** [comment]
> @Adbib I fixed up the docs. I added my suggestions too. Are we considering all the cases for `isActive`prop? what about href="/about/" is it considered?

yes is it

**Comment by gioboa** [comment]
I added the CodeSandbox example and it doesn't seem to work and I can't understand how verify isPending 🤔 
Screenshot 2023-12-26 at 17 26 54

**Comment by Adbib** [comment]
> I added the CodeSandbox example and it doesn't seem to work and I can't understand how verify isPending 🤔 Screenshot 2023-12-26 at 17 26 54

I think because we do a full-page refresh with a tag.
if you try it with < Link > that does client-side navigation it should work

**Comment by gioboa** [comment]
Can you verify the missing active and pending classes pls?
If your implementation is with Link you can change it

**Comment by Adbib** [comment]
> Can you verify the missing active and pending classes pls? If your implementation is with Link you can change it

its work now

**Comment by gioboa** [comment]
I can't verify it now, can you send me a screenshot pls?

**Comment by Adbib** [comment]
> I can't verify it now, can you send me a screenshot pls?

this video with implementation of the same code : 
https://github.com/BuilderIO/qwik/assets/17273701/61f42dec-578e-4174-8cea-00af5e2e2d06

**Review by gioboa — APPROVED** [review]

**Comment by gioboa** [comment]
@Adbib I cleaned up the example. I removed the pending state b/c I wan't able to make it works.
If for you it's ok I will merge it.

**Comment by Adbib** [comment]
> @Adbib I cleaned up the example. I removed the pending state b/c I wasn't able to make it works. If for you it's ok I will merge it.

okay lets merge it for now untiil we find a way to pendding

**Pull request merged**